src/coreclr/vm/stubmgr.cpp

```diff
@@ -1188,39 +1188,58 @@ BOOL StubLinkStubManager::DoTraceStub(PCODE stubStartAddress,
     TADDR pRealAddr = 0;
     if (stub->IsMulticastDelegate())
     {
-        LOG((LF_CORDB, LL_INFO10000,
-             "StubLinkStubManager(MCDel)::DoTraceStub: stubStartAddress=0x%08x\n",
-             stubStartAddress));
-
-        LOG((LF_CORDB, LL_INFO10000,
-             "StubLinkStubManager(MCDel)::DoTraceStub: stub=0x%08x MGR_PUSH to entrypoint:0x%x\n", stub,
-             stub->GetEntryPoint()));
-
         // If it's a MC delegate, then we want to set a BP & do a context-ful
         // manager push, so that we can figure out if this call will be to a
         // single multicast delegate or a multi multicast delegate
         trace->InitForManagerPush(stubStartAddress, this);
-
+        LOG_TRACE_DESTINATION(trace, stubStartAddress, "StubLinkStubManager(MCDel)::DoTraceStub");
         return TRUE;
     }
-    else if (stub->GetPatchOffset() == 0)
+    else if (stub->IsInstantiatingMethodStub())
     {
-        LOG((LF_CORDB, LL_INFO10000, "StubLinkStubManager::DoTraceStub: patch offset is 0!\n"));
-
-        return FALSE;
+        trace->InitForManagerPush(stubStartAddress, this);
+        LOG_TRACE_DESTINATION(trace, stubStartAddress, "StubLinkStubManager(InstantiatingMethod)::DoTraceStub");
+        return TRUE;
     }
-    else
+    else if (stub->GetOffset() != 0)
```

Comment:
It isn't obvious to me that a patch offset of 0 is inherently bad. If there is somewhere in the code that sets up this invariant a comment to describe or point at it would be very handy.

Reply:
This is specifically for x86 Delegates where `!defined(FEATURE_MULTICASTSTUB_AS_IL)`. This is only ever non-zero in that scenario. I can add a comment here for x86 if desired?